common/models/user.js

```diff
@@ -580,7 +580,10 @@ module.exports = function(User) {
             query && query.redirect !== undefined ? query.redirect :
             undefined;
 
-          ctx.res.redirect(redirectUrl);
+          if (redirectUrl !== undefined) {
+            ctx.res.location(redirectUrl);
+          }
+          next()
         } else {
           next(new Error('transport unsupported'));
```

Comment:
Can you, please, tell, why should I rise `transport unsupported` error if `redirectUrl` is passed?